catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,184 @@
+        queueCapacity > 0,
+        String.format("Iceberg metrics queue capacity:%d should greater than 0", queueCapacity));
+    queue = new LinkedBlockingQueue(queueCapacity);
+    metricsWriterThread =
+        new Thread(
+            () -> {
+              while (Thread.currentThread().isInterrupted() == false) {
+                MetricsReport metricsReport;
+                try {
+                  metricsReport = queue.take();
+                } catch (InterruptedException e) {
+                  LOG.warn("Iceberg Metrics writer thread is interrupted");
+                  return;
+                }
+                if (metricsReport != null) {
+                  doSave(metricsReport);
+                }
+              }
+
+              MetricsReport metricsReport = queue.poll();
+              while (metricsReport != null) {
+                logMetrics(
+                    "Drop Iceberg metrics because it's time to close metrics storage.",
+                    metricsReport);
+                metricsReport = queue.poll();
+              }
+            });
+    metricsWriterThread.setName("Iceberg-metrics-writer");
+    metricsWriterThread.setDaemon(true);
+  }
+
+  private IcebergMetricsStorage loadIcebergMetricsStorage(String metricsStorageType) {
+    String metricsStorageClass =
+        ICEBERG_METRICS_STORAGES.getOrDefault(metricsStorageType, metricsStorageType);
+    try {
+      return (IcebergMetricsStorage)
+          Class.forName(metricsStorageClass).getDeclaredConstructor().newInstance();
+    } catch (Exception e) {
+      LOG.error(
+          "Failed to create and initialize Iceberg metrics storage by name {}.",
+          metricsStorageType,
+          e);
+      throw new RuntimeException(
+          "Failed to create and initialize Iceberg metrics storage : " + metricsStorageType, e);
+    }
+  }
+
+  private void logMetrics(String message, MetricsReport metricsReport) {
+    LOG.info("{} {}", message, icebergMetricsFormatter.toPrintableString(metricsReport));
+  }
+
+  public void start() {
+    if (metricsWriterThread != null) {
+      metricsWriterThread.start();
+    }
+
+    if (scheduledThreadPoolExecutor != null) {
+      scheduledThreadPoolExecutor.scheduleAtFixedRate(
+          () -> {
+            Instant now = Instant.now();
+            Instant expireTime = IcebergRestUtils.calculateNewTimestamp(now, -24 * retainDays);
+            LOG.info("Try clean Iceberg expired metrics, {}", expireTime);
+            try {
+              storage.clean(expireTime);
+            } catch (Exception e) {
+              LOG.warn("Clean Iceberg metrics failed,", e);
+            }
+          },
+          0,
+          1,
+          TimeUnit.HOURS);
+    }
+  }
+
+  private void doSave(MetricsReport metricsReport) {
+    try {
+      storage.save(metricsReport);
+    } catch (Exception e) {
+      LOG.warn("Write Iceberg metrics failed,", e);
+    }
+  }
+
+  public void save(MetricsReport metricsReport) {
```

Comment:
Where did you call this method?

Reply:
IcebergTableOperations#reportTableMetrics()

Reply:
This save method is quite confused with the method above, you'd better choose a better name.

Also you should put all the public method together, then all all private method.

Reply:
How about using record?